src/app/services/api/api.service.ts

```diff
@@ -98,38 +158,57 @@ export class ApiService {
    * Gets information about the current coin supply.
    */
   getCoinSupply(): Observable<GetCoinSupplyResponse> {
-    return this.get('coinSupply');
+    const url = !this.nodeUrl() ? 'coinSupply' : 'v1/coinSupply';
+
+    return this.get(url);
   }
 
   /**
    * Gets the list of unspent outputs of an address.
    * @param address Address to consult.
    */
   getCurrentBalance(address: string): Observable<GetCurrentBalanceResponse> {
-    return this.get('currentBalance', { addrs: address });
+    const url = !this.nodeUrl() ? 'currentBalance' : 'v1/outputs';
+
+    return this.get(url, { addrs: address });
   }
 
   /**
    * Gets the balance of an address.
    * @param address Address to consult.
    */
   getBalance(address: string): Observable<GetBalanceResponse> {
-    return this.get('balance', { addrs: address });
+    const url = !this.nodeUrl() ? 'balance' : 'v1/balance';
+
+    return this.get(url, { addrs: address });
   }
 
   /**
    * Gets a transaction by its hash.
    * @param transactionId Transaction hash.
    */
   getTransaction(transactionId: string): Observable<GetTransactionResponse> {
-    return this.get('transaction', { txid: transactionId, verbose: 1 });
+    const url = !this.nodeUrl() ? 'transaction' : 'v1/transaction';
+
+    return this.get(url, { txid: transactionId, verbose: 1 });
   }
 
   /**
    * Gets the list of unlocked addresses with most coins.
    */
   getRichlist(): Observable<RichlistEntry[]> {
-    return this.get('richlist').pipe(map((response: any) => response.richlist));
+    const url = !this.nodeUrl() ? 'richlist' : 'v1/richlist';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin-explorer/pullRequest?prid=7163626)
```suggestion
    const url = !this.nodeUrl() ? "richlist" : 'v1/richlist';
```